src/GovUk.Education.ExploreEducationStatistics.Data.Services/TimePeriodService.cs

```diff
@@ -48,6 +50,26 @@ public IEnumerable<(int Year, TimeIdentifier TimeIdentifier)> GetTimePeriods(Gui
             return TimePeriodUtil.GetTimePeriodRange(start, end);
         }
 
+        public TimePeriodRangeLabels GetTimePeriodRangeLabels(Guid subjectId)
+        {
+            var observationsQuery = _context
+                .Observation
+                .Where(observation => observation.SubjectId == subjectId);
+            var orderedTimePeriods = GetDistinctObservationTimePeriods(observationsQuery).ToList();
+
+            if (!orderedTimePeriods.Any())
+            {
+                return new TimePeriodRangeLabels();
+            }
+
+            var first = orderedTimePeriods.First();
+            var last = orderedTimePeriods.Last();
+
+            return new TimePeriodRangeLabels(
+                TimePeriodLabelFormatter.Format(first.Year, first.TimeIdentifier),
```

Comment:
Out of interest, why do we pass in Year as the first argument?  Don't some time periods operate as just weeks i.e. without any Year component?  I can see from your tests that the Weeks behave themselves - just curious about this argument.

Reply:
I think all time identifiers are always associated with a year. So with weeks the time period range labels could be from / to something like "2020 Week 12", "2021 Week 2".